### PR TITLE
Bump pydantic version

### PR DESCRIPTION
**Pull request by maciejmajerczyk**
Issue: https://github.com/evidentlyai/evidently/issues/862

Notes:

upgraded pydantic version to allow versions <3
fixed tests in test_pydantic_utils.py yo use _pydantic_compat.py
increased evidently version so we get a new version

### CODE REVIEW

File: src/evidently/_version.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
 # coding: utf-8
 
-version_info = (0, 4, 13)
+version_info = (0, 4, 14)
```
[the remark this answers is not in the document]
Comment (maciejmajerczyk) [reply src/evidently/_version.py]: updated

**Comment by DimaAmega** [comment]
Hi @maciejmajerczyk! Thanks for your contribution!

As you can see, there is a failed [test](https://github.com/evidentlyai/evidently/actions/runs/7465515418/job/20384723541)

To reproduce this you should run 
```bash
evidently ui --port 8000 --demo-projects all
```
Next go to http://localhost:8000 and see logs in terminal

**Comment by mike0sv** [comment]
https://github.com/tiangolo/fastapi/issues/10360

**Comment by emeli-dral** [comment]
Hi @maciejmajerczyk ,
I'm sorry for the late reply.

We discussed the issue with the pydantic version. Indeed this should be addressed immediately and here is an option we could suggest now: as evidently ui service depends of FastAPI we cannot bump pydantic version right now (https://github.com/tiangolo/fastapi/issues/10360). However, if you use evidently to generate Reports or TestSuites only (without running ```evidently ui```) or if you calculate Reports or TestSuites in one environment but run  evidently ui service on the other environment you can do the following. It is ok to bump pydantic version after you installed evidently in the environment where you do not need to run  ```evidently ui```. Report or TestSuite calculations work just fine with pydantic 2.

We certainly understand that this solution is neither convenient nor stable. This is why we now replace FastAPI with [litestar](https://docs.litestar.dev/2/index.html). After this replacement we will be able to bump pydantic version. I believe it will happen in a couple of weeks.

**Pull request closed without merging**